uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveBalanceChangeTest.php

```diff
@@ -721,6 +738,16 @@ public function testLeaveRequestBalanceForEntitlementShouldIncludeTOILBalanceCha
       'to_date' => CRM_Utils_Date::processDate('+10 days')
     ], true);
 
+    $leaveRequest = LeaveRequestFabricator::fabricateWithoutValidation([
+      'contact_id' => $entitlement->contact_id,
+      'type_id' => $entitlement->type_id,
+      'from_date' => CRM_Utils_Date::processDate('+11 days'),
+      'to_date' => CRM_Utils_Date::processDate('+11 days')
+    ], true);
+
+    //The leave request is soft deleted and therefore will nnot be accounted for in the calculation
+    LeaveRequest::softDelete($leaveRequest->id);
+
```

Comment:
Same thing here and for the other tests. Probably better to create a separate test for testing things related to deleted records